plugins/optimization-detective/storage/class-od-storage-lock.php

```diff
@@ -86,7 +86,13 @@ public static function get_ttl(): int {
 	 * @return non-empty-string Transient key.
 	 */
 	public static function get_transient_key(): string {
-		$ip_address = $_SERVER['HTTP_X_FORWARDED_FOR'] ?? $_SERVER['REMOTE_ADDR'];
+		if ( isset( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
+			$ip_address = sanitize_text_field( wp_unslash( $_SERVER['HTTP_X_FORWARDED_FOR'] ) );
+		} elseif ( isset( $_SERVER['REMOTE_ADDR'] ) ) {
+			$ip_address = sanitize_text_field( wp_unslash( $_SERVER['REMOTE_ADDR'] ) );
```

Comment:
This is unnecessary since it is sent into `wp_hash()`, but since that function technically is pluggable, it's better to ensure that `$ip_address` is safe.

Reply:
I guess `rest_is_ip_address()` could be included here as well, but that would be even more over the top, maybe.